ethers-middleware/src/proxy_wallet/ds_proxy/mod.rs

```diff
@@ -0,0 +1,89 @@
+use super::{ProxyWallet, ProxyWalletError};
+use ethers_contract::BaseContract;
+use ethers_core::{
+    abi::{parse_abi, Tokenize},
+    types::*,
+};
+
+const DS_PROXY_EXECUTE_SIGNATURE: &str =
+    "function execute(address target, bytes data) public payable returns (bytes memory response)";
+
+#[derive(Debug)]
+pub struct DsProxy {
+    address: Address,
+}
+
+impl DsProxy {
```

Comment:
Could be cool to add a few other convenience methods, e.g. a `create` method which allows using the DsProxyRegistry to create a new proxy easily